src/compiler/commandLineParser.ts

```diff
@@ -855,6 +855,15 @@ const commandOptionsWithoutBuild: CommandLineOption[] = [
         affectsBuildInfo: true,
         affectsSemanticDiagnostics: true,
     },
+    {
+        name: "erasableSyntaxOnly",
+        type: "boolean",
+        category: Diagnostics.Interop_Constraints,
+        description: Diagnostics.Do_not_allow_runtime_constructs_that_are_not_part_of_ECMAScript,

```

Comment:
JSX?

Reply:
Open to suggestions 🤷

Reply:
I'd vote no since that would curse anyone trying to use this for "purist" purposes to not being able to use JSX at all; you're already going to get an error from Node, and in other runtimes or with a loader, it might even work just fine....

Reply:
Yeah, the fact that you already have to put JSX content in a `.?sx` file means you can't possibly make the mistake of "not realizing enum is special" or whatever.

I thought Daniel wanted the text to say something different, which technically JSX *is* a non-ES runtime construct, but the phrasing would maybe get super awkward

Reply:
Ohh that interpretation makes a lot more sense, oops

Reply:
Overall I think it is fine either way, just a slightly weird distinction.